src/bridge/mod.rs

```diff
@@ -236,25 +252,52 @@ async fn launch(handler: NeovimHandler, grid_size: Option<Dimensions>) -> Result
     res.map(|()| session)
 }
 
-async fn run(session: NeovimSession) {
-    match session.io_handle.await {
-        Err(join_error) => error!("Error joining IO loop: '{}'", join_error),
-        Ok(Err(error)) => {
-            if !error.is_channel_closed() {
-                error!("Error: '{}'", error);
+async fn run(session: NeovimSession, proxy: EventLoopProxy<UserEvent>) {
+    let mut session = session;
+
+    if let Some(process) = session.neovim_process.as_mut() {
+        let neovim_exited = select! {
+            _ = &mut session.io_handle => false,
+            _ = process.wait() => {
+                log::info!("The Neovim process quit before the IO stream, waiting two seconds");
+                true
+            }
+        };
```

Comment:
I tried this approach as one of my first attempts when trying to fix this problem, but `wait()` doesn't detect that the process is gone.

Reply:
On Linux this workaround is needed for Neovim 0.9.5 without my fix https://github.com/neovim/neovim/pull/28157. There it correctly detects both the stdio stream close and process exit.